### ISSUE TITLE
Implement "incremental ready" build process 

### ISSUE THREAD

**Issue by desigonz** [cityofaustin/techstack#3188]
Joplin PR: https://github.com/cityofaustin/joplin/pull/406
Janis PR: https://github.com/cityofaustin/janis/pull/559
Deployed Joplin: https://joplin-pr-3151-urls-in-api.herokuapp.com
Deployed Janis: https://janis-3152-riffin.netlify.com

## QA Notes
* Top links for topics are now managed on topic pages. Topic collection pages rely on this to generate the topic cards. When top links have not been set the cards look weird and broken:
  * [topic collection page with top links set on topics](https://janis-3152-riffin.netlify.com/en/housing-utilities/recycling-trash-and-compost/)
  * [topic collection page with no top links set](https://janis-3152-riffin.netlify.com/en/health-safety/health-records-certificates-2/)
* view live links go to janis.austintexas.io (would be nice to have those change based on branch settings for PR apps, do we have an issue for that?)

### Old notes
  - Joplin should expose an API endpoint for:
    - The information architecture/url structure of the site
    - Each page by global ID with enough information returned to build the whole page, this includes all children/top links/contextual nav info etc.
      - These can vary by url, How/where we want to handle that? Can we build some of it into JanisBasePage?
  - Janis should:
    - Request a list of pages to build, and get a list of url/id pairs.
    - Build a page at each url by requesting data by id
  - Once this is complete, a release branch should be created off of the main incremental builds epic branch. QA can then be performed on this release branch, and once everything passes, we can merge to staging/prod. Even though this isn't incremental yet, it is quite possible we can speed up builds with better parallelization.

**Comment by desigonz** [comment]
Will updating https://joplin-pr-3151-urls-in-api.herokuapp.com publish out to https://janis-3152-riffin.netlify.com/?

**Comment by desigonz** [comment]
Your comment `would be nice to have those change based on branch settings for PR apps, do we have an issue for that?` is something @easherma worked on today, at least for the PR branch affiliated with #3053...

**Comment by briaguya** [comment]
> Will updating https://joplin-pr-3151-urls-in-api.herokuapp.com publish out to https://janis-3152-riffin.netlify.com/?

YES! Thanks @Niicck!

**Comment by desigonz** [comment]
I'm starting my QA process now and will just be listing a bunch of things in a string of comments below. Later @briaguya and I can look them over together and prioritize.

**Comment by desigonz** [comment]
**Describe the bug**
Janis—"Departments" page (not the content type)
"Austin Public Health" card is displaying weird and longer than the others
![image](https://user-images.githubusercontent.com/5839508/67495483-96cf0400-f640-11e9-8463-04c8bcc313b0.png)

**Additional comments**
This is probably totally unrelated to this issue :) but documenting here for now, will split into new issue later

**Comment by briaguya** [comment]
@desigonz re: "Departments" page, looks like there are two issues there:
1. I forgot to only show departments that are live! (just checked in a fix for that)
2. We need the links/buttons/tiles to look uniform/good (new issue would be wonderful for that!)

re: Topic Cards on topic collections
our logic before was show 4 on every card, starting with "top links" and then continuing on with any other page (seemingly at random) after that.

My hope is we:
* set a verbal policy to not publish a topic without any top links for now
  * maybe once we set up separate publish/draft logic we could turn this into a publish requirement?
* maybe design a content author friendly flow for automatically setting top links in Joplin when adding pages to topics that don't have any?
  * I like the idea of a content author being able to see and manage exactly what is going to show up on any given card, instead of having some janis magic filling it in.

**Comment by desigonz** [comment]
@briaguya Tell me more about your recommendation for a verbal policy now, rather than implementing some guardrails. What does the dev complexity look like there? Or is your recommendation entirely coming from the UX experience perspective on your last bullet point?

Also, I think since you pushed up some code changes, publisher is no longer working, you might need to reset it so that Janis will build.

**Comment by desigonz** [comment]
### Choosing a top links on a topic
- [ ] Right now it shows all live non-IA pages. This seems potentially problematic, because I can choose a page that has been assigned this topic (let's call it topic Z), and when I go to that page, it doesn't reflect that it's in Topic Z, only the topics it was previously assigned to (if any). Will this break things? Maybe it should only show pages already assigned to this topic.
- [ ] Need to update the microcopy:
  - [ ] `Choose content to feature on this topic`
  - [ ] `Only the following published page types within this topic may be chosen for this field: Information page, Service page, Guide page, Official document page. Search results will exclude pages of other types.`

![image](https://user-images.githubusercontent.com/5839508/67514094-bde9fd80-f661-11e9-8422-9aaafaa3107d.png)

**Comment by briaguya** [comment]
@desigonz re: verbal policy.

This is based on an assumed flow of:
* all top links for a topic are content managed, no surprises with extra links in there.
* ideally, we would prevent publishing a topic without top links
  * This causes problems before we have separate publish/draft save logic, because we'd get conflicts around adding a page to a topic/making it top/being able to save the topic in the first place

re: choosing top links
> because I can choose a page that has been assigned this topic

I'm assuming you mean hasn't. This is also how our top links and related links on department pages work now (no filter by dept). Probably worth making a separate issue for.

**Comment by briaguya** [comment]
@easherma re: @desigonz's microcopy changes:

looks like the logic for that line is happening in https://github.com/cityofaustin/joplin/blob/cf0c03cb25ba7f26cfff5ca038a4498f466e7329/joplin/templates/wagtailadmin/chooser/browse.html#L11

I'd consider this a bug in how we're determining the choose_str logic there, as there will be non department situations where we want to choose all 4 non IA page types.

**Comment by briaguya** [comment]
@desigonz I just merged master into these branches (including Eric's link stuff that just got merged in). This means a few things:
1. I fixed the build problems (found what was causing your issue and checked it in)
2. Previews are working using the branch stuff now!

re: top links again
I did notice a side effect of the choosing top links bug: when getting the url for the top linked page it assumes it is already a child of the topic so we get a url that gives us a 404. This is definitely something that will happen if we assign a page as a top/related link for a department without setting it as related to that department as well. My vote is still to make this another issue

**Comment by desigonz** [comment]
Brian, here's @ablangworthy27's assignments for top links: https://github.com/cityofaustin/techstack/issues/3259#issuecomment-547074961

I believe Jo is queueing up some new APH content to publish soon—might have more to consider then.

**Comment by briaguya** [comment]
Before merging to prod: note on what our autogenerated top links are:

* Household waste
  * Get your bulk items collected
  * Get ready for curbside compost
  * Drop off household hazardous waste and other recyclables
  * Look up your trash, recycling, and compost pickup days
* Compost and food waste
  * Get ready for curbside compost
  * Set out your leaves, grass, and branches for pickup
* Recycling and reuse
  * Pick up free paint and other household items
  * Look up your trash, recycling, and compost pickup days
* Birth and death certificates
  * Get a birth or death certificate
* Disease prevention
  * Find free condoms and learn other ways to prevent HIV
  * Get immunizations for you and your family
  * Measles in Austin
* Disaster preparedness
  * Prepare for public health emergencies
* Police oversight and feedback
  * File a complaint about an Austin police officer
  * Thank an Austin police officer
  * Official complaint and discipline documents

**Comment by briaguya** [comment]
@desigonz this is in prod now, I went ahead and added top links to the topics as per my previous comment
